src/test/java/io/jenkins/plugins/audit/listeners/BuildListenerTest.java

```diff
@@ -36,14 +37,20 @@ public void teardown() {
         app.clear();
     }
 
-    @Issue("JENKINS-55608")
+    @Issue("JENKINS-55608, JENKINS-56645")
     @Test
-    public void testAuditOnBuildStart() throws Exception{
+    public void testAuditOnBuildStartAndFinish() throws Exception{
         List<LogEvent> events = app.getEvents();
 
         project.getBuildersList().add(new Shell("echo Test audit-log-plugin"));
         project.scheduleBuild2(0).get();
 
-        assertEquals("Event on build start not triggered", 1, events.size());
+        /*
+        Order of expected events
+        1) Start of the Build
+        2) Finish of the Build
+         */
+
+        assertEquals("Events on build start and complete not triggered", 2, events.size());
```

Comment:
This test can work for now, though we'll still need to develop better assertions to ensure the ordering of log messages and that the contents are being logged properly.

Reply:
Sure. I wish the selected candidate (or a newbie task doer) can take it up with refactoring the tests that may have to change with adding events. I looked for some ideas to assert in LogEvent entity and only then added them as comments.